File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/Exec.py
#!/usr/bin/python

### Exec.py

import os                                                           #(A)   

environ_vars = os.environ.keys()                                    #(B)
environ_vars.sort()                                                 #(C)
print environ_vars                                                  #(D)

os.environ['PATH'] += ':/home/kak/myscripts/'                       #(E)

os.environ['ACK_MSG'] = "You said: ";                               #(F)

os.execv('/bin/sh', ['sh', '-c', 'while a=a; do read MYINPUT;    \
                     echo $ACK_MSG $MYINPUT; done'] )               #(G)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/Exec.pl
#!/usr/bin/perl -w

### Exec.pl

use strict;

print map {$_ . " "} sort keys %ENV;                               #(A)
print "\n\n";                 

$ENV{PATH} .= '/home/kak/myscripts/';                              #(B)

$ENV{ACK_MSG} = "You said: ";                                      #(C)

exec('while a=a; do read MYINPUT; echo $ACK_MSG $MYINPUT; done');  #(D)
